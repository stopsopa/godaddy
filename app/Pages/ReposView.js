
import React from 'react';

import {
  NavLink,
  Link,
  Redirect,
  useParams,
} from 'react-router-dom';

import Menu from './Menu'

export default function ReposView({
  list,
  page,
  pages,
}) {
  return (
    <Menu>
      {list ? (
        <>
          <div className="repo-list">
            <div className="list">
              {list.map(repo => (
                <div key={repo.node_id} className="repo">
                  <Link className="name" to={`/repo/${repo.name}`}>{repo.name}</Link>
                  <div className="description">{repo.description}</div>
                </div>
              ))}
            </div>
            <div className="pagination">
              {(function () {
                const buttons = [];
                for (let i = 1 ; i <= pages ; i += 1 ) {
                  buttons.push(i);
                }
                return buttons;
              }()).map(i => (<NavLink key={i} to={`/repos/${i}`} activeClassName="active">{i}</NavLink>))}
            </div>
          </div>
        </>
      ) : (
        <div>Loading...</div>
      )}
    </Menu>
  );
}


