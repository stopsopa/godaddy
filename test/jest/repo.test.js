
import React from 'react';

const {
  fetchJson,
} = require('../fetchJsonForTests');

import { render, screen } from '../test-utils'

import ReposView from '../../app/Pages/ReposView'

describe('repo', () => {

  test('loading', async done => {

    render(<ReposView />);

    expect(document.body.innerHTML).toMatchSnapshot();

    done();
  });
});

describe('repo', () => {

  test('repo component', async done => {

    const json = await fetchJson(`https://api.github.com/orgs/godaddy/repos`);

    render(<ReposView
      list={json.body}
      page={1}
      pages={5}
    />);

    expect(document.body.innerHTML).toMatchSnapshot();

    done();
  });
});