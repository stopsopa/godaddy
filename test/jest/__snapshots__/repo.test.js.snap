// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`repo loading 1`] = `"<div><div><a href=\\"/\\">Homepage</a><hr><div>Loading...</div></div></div>"`;

exports[`repo repo component 1`] = `"<div><div><a href=\\"/\\">Homepage</a><hr><div class=\\"repo-list\\"><div class=\\"list\\"><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/gdapi-php\\">gdapi-php</a><div class=\\"description\\">A PHP client for Go Daddy® REST APIs</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/lazy-social-buttons\\">lazy-social-buttons</a><div class=\\"description\\">A JavaScript plugin to place social buttons on a page on user interaction (mouseover) to spare the initial page load from the 300kb+ download requests for social APIs.</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/gdapi-csharp\\">gdapi-csharp</a><div class=\\"description\\">A C# client for Go Daddy® REST APIs</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/gdapi-ui\\">gdapi-ui</a><div class=\\"description\\">An in-browser client for Go Daddy® REST APIs</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/gdapi-python\\">gdapi-python</a><div class=\\"description\\">A Python client for Go Daddy® REST APIs</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/node-cluster-service\\">node-cluster-service</a><div class=\\"description\\">Turn your single process code into a fault-resilient, multi-process service with built-in REST &amp; CLI support. Restart or hot upgrade your web servers with zero downtime or impact to clients.</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/node-redis-ha\\">node-redis-ha</a><div class=\\"description\\">Redis high-availability client library for node</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/better_spree_paypal_express\\">better_spree_paypal_express</a><div class=\\"description\\">(An attempt at) A better Spree PayPal Express Extension. Currently in beta.</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/spree\\">spree</a><div class=\\"description\\">Spree is a complete open source e-commerce solution for Ruby on Rails.</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/knockout\\">knockout</a><div class=\\"description\\">Knockout makes it easier to create rich, responsive UIs with JavaScript</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/jquery-binder\\">jquery-binder</a><div class=\\"description\\">jQuery Plugin to track elements with a given event listener.</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/node-access-log\\">node-access-log</a><div class=\\"description\\">Add simple access logs to any http or https server</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/node-http-cache\\">node-http-cache</a><div class=\\"description\\">An extensible caching interface for HTTP traffic.</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/node-http-cache-cassandra\\">node-http-cache-cassandra</a><div class=\\"description\\">A Cassandra provider for the extensible HTTP caching library http-cache.</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/apartment\\">apartment</a><div class=\\"description\\">Database multi-tenancy for Rack (and Rails) applications</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/gdapi-perl\\">gdapi-perl</a><div class=\\"description\\">A Perl client for Go Daddy® REST APIs</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/node-gd-assets\\">node-gd-assets</a><div class=\\"description\\">CSS, JS, and Handlebars combiner, compressor, and server</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/node-soap\\">node-soap</a><div class=\\"description\\">A SOAP client and server for node.js.</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/node-callback-pool\\">node-callback-pool</a><div class=\\"description\\">Simple utility to allow callbacks to stack up until the drain has been unplugged.</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/spree_gateway\\">spree_gateway</a><div class=\\"description\\">Community maintained collection of gateways for use in Spree</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/node-priam\\">node-priam</a><div class=\\"description\\">A simple Cassandra driver for NodeJS. It wraps node-cassandra-cql with additional error/retry handling, external .cql file support, and connection option resolution from an external source.</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/spree_auth_devise\\">spree_auth_devise</a><div class=\\"description\\">Provides authentication for Spree by using Devise.</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/node-soap-client-utils\\">node-soap-client-utils</a><div class=\\"description\\"></div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/spree_weight_based_shipping_calculator\\">spree_weight_based_shipping_calculator</a><div class=\\"description\\">Spree extension for weight-based shipping calculation</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/GDRouting\\">GDRouting</a><div class=\\"description\\"></div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/python-oauth2\\">python-oauth2</a><div class=\\"description\\">A fully tested, abstract interface to creating OAuth clients and servers.</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/orm-cassandra-adapter\\">orm-cassandra-adapter</a><div class=\\"description\\"></div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/winston-newrelic\\">winston-newrelic</a><div class=\\"description\\">Reports winston log errors to new-relic</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/openstack-puppet\\">openstack-puppet</a><div class=\\"description\\">Go Daddy Puppet manifests for installing, configuring, and managing OpenStack components</div></div><div class=\\"repo\\"><a class=\\"name\\" href=\\"/repo/openstack-ansible\\">openstack-ansible</a><div class=\\"description\\">Ansible playbooks for managing OpenStack infrastructure</div></div></div><div class=\\"pagination\\"><a href=\\"/repos/1\\">1</a><a href=\\"/repos/2\\">2</a><a href=\\"/repos/3\\">3</a><a href=\\"/repos/4\\">4</a><a href=\\"/repos/5\\">5</a></div></div></div></div>"`;
